src/cluster.c

```diff
@@ -4597,44 +4637,59 @@ NULL
         /* Check that all the arguments are parseable and that all the
          * slots are not already busy. */
         for (j = 2; j < c->argc; j++) {
-            if ((slot = getSlotOrReply(c,c->argv[j])) == -1) {
+            if ((slot = getSlotOrReply(c,c->argv[j])) == C_ERR) {
                 zfree(slots);
                 return;
             }
-            if (del && server.cluster->slots[slot] == NULL) {
-                addReplyErrorFormat(c,"Slot %d is already unassigned", slot);
+            if(getSlotStatus(c, slots, del, slot, slot) == C_ERR){
                 zfree(slots);
                 return;
-            } else if (!del && server.cluster->slots[slot]) {
-                addReplyErrorFormat(c,"Slot %d is already busy", slot);
+            }
+        }
+        addAndDelSlotOperation(c, slots, del);    
+        zfree(slots);
+        clusterDoBeforeSleep(CLUSTER_TODO_UPDATE_STATE|CLUSTER_TODO_SAVE_CONFIG);
+        addReply(c,shared.ok);
+    } else if ((!strcasecmp(c->argv[1]->ptr,"addslotsrange") ||
+               !strcasecmp(c->argv[1]->ptr,"delslotsrange")) && c->argc >= 4) {
+        if(c->argc % 2 == 1){
+            addReplyErrorFormat(c,"Missed one endslot");
```

Comment:
For other commands, this error message usually looks like this:

```suggestion
            addReplyErrorFormat(c, "wrong number of arguments for '%s' command",
                                c->cmd->name);
```

Reply:
Perhaps it should be "odd number of arguments ...".

Reply:
I'm going to agree with @zuiderkwast, I think just saying that the number of arguments is wrong is the best indicator they should take a step back and make sure they sent the right command.

Reply:
If we want these error messages to be consistent, we should have a macro or function for it. For modules, there's a RedisModule_WrongArity function IIRC.

Reply:
@madolson @zuiderkwast address your comments, thanks

Reply:
@zuiderkwast Agree about the standard function. let's do the refactor after this is merged.